From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Sat, 2 Aug 2014 23:43:26 +0200
Subject: disable some libopus feature for ARCH < ARMv6

BZ#1025689
https://bugzilla.mozilla.org/show_bug.cgi?id=1025689

This patch here based on Mike's work for Iceweasel. Big thank to Mike
who figured out the issue.

Closes: #754633
---
 mozilla/media/libopus/moz.build | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/mozilla/media/libopus/moz.build b/mozilla/media/libopus/moz.build
index 4a0bb59..0e4f6a5 100644
--- a/mozilla/media/libopus/moz.build
+++ b/mozilla/media/libopus/moz.build
@@ -23,10 +23,11 @@ if CONFIG['CPU_ARCH'] == 'arm' and CONFIG['GNU_AS']:
     DEFINES['OPUS_ARM_ASM'] = True
     DEFINES['OPUS_ARM_EXTERNAL_ASM'] = True
     DEFINES['OPUS_ARM_INLINE_ASM'] = True
-    DEFINES['OPUS_ARM_INLINE_EDSP'] = True
-    DEFINES['OPUS_ARM_MAY_HAVE_EDSP'] = True
-    DEFINES['OPUS_ARM_MAY_HAVE_MEDIA'] = True
-    DEFINES['OPUS_ARM_MAY_HAVE_NEON'] = True
+    if int(CONFIG['ARM_ARCH']) >= 6:
+         DEFINES['OPUS_ARM_INLINE_EDSP'] = True
+         DEFINES['OPUS_ARM_MAY_HAVE_EDSP'] = True
+         DEFINES['OPUS_ARM_MAY_HAVE_MEDIA'] = True
+         DEFINES['OPUS_ARM_MAY_HAVE_NEON'] = True
 
 if CONFIG['MOZ_DEBUG']:
     DEFINES['ENABLE_ASSERTIONS'] = True
